src/main/java/com/google/api/codegen/metacode/InitFieldConfig.java

```diff
@@ -69,11 +72,16 @@ public boolean hasFormattedInitValue() {
     return entityName() != null && value() != null;
   }
 
-  private static String parseValueString(String valueString, String stringToHash) {
+  private static InitValue parseValueString(String valueString, String stringToHash) {
+    InitValue initValue = new InitValue(valueString, InitValueType.Literal);
     if (valueString.contains(randomValueToken)) {
       String randomValue = Integer.toString(Math.abs(stringToHash.hashCode()));
       valueString = valueString.replace(randomValueToken, randomValue);
+      initValue = new InitValue(valueString, InitValueType.Literal);
+    } else if (valueString.contains(projectIdToken)) {
+      valueString = projectIdVariableName;
```

Comment:
This replaces the whole string, which is different from the randomValue case. Is it possible to allow the project id to be specified alongside other data? If not, it should be an error to specify a value which contains projectIdToken, but is not equal to projectIdToken, because the extra is lost.